From a9ff39087092b21059fca046ace9ca87770692a4 Mon Sep 17 00:00:00 2001
From: Nobuyoshi Nakada <nobu@ruby-lang.org>
Date: Tue, 29 Sep 2020 13:41:59 +0900
Subject: [PATCH] Fixed installation failure [Bug #17191]

Try update and extract bundled gems only when baseruby is
available.  It should be done only when installing from
developemental build and not from the tarball, but it is not
obvious to differentiate them.
---
 common.mk     | 2 +-
 defs/gmake.mk | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git common.mk common.mk
index 603429bd35..509f09c1fe 100644
--- common.mk
+++ common.mk
@@ -1296,7 +1296,7 @@ update-config_files: PHONY
 	    config.guess config.sub
 
 refresh-gems: update-bundled_gems prepare-gems
-prepare-gems: update-gems extract-gems
+prepare-gems: $(HAVE_BASERUBY:yes=update-gems) $(HAVE_BASERUBY:yes=extract-gems)
 
 update-gems$(gnumake:yes=-nongnumake): PHONY
 	$(ECHO) Downloading bundled gem files...
diff --git defs/gmake.mk defs/gmake.mk
index 7e9566fee6..31af44a4f1 100644
--- defs/gmake.mk
+++ defs/gmake.mk
@@ -246,7 +246,7 @@ HELP_EXTRA_TASKS = \
 	"  update-github:       merge master branch and push it to Pull Request [PR=1234]" \
 	""
 
-extract-gems: update-gems
+extract-gems: $(HAVE_BASERUBY:yes=update-gems)
 
 BUNDLED_GEMS := $(shell sed '/^[ 	]*\#/d;/^[ 	]*$$/d;s/[ 	][ 	]*/-/;s/[ 	].*//' $(srcdir)/gems/bundled_gems)
 
-- 
2.28.0

